packages/twenty-front/src/modules/object-record/object-options-dropdown/components/ObjectOptionsDropdownContent.tsx

```diff
@@ -0,0 +1,184 @@
+import { TableOptionsHotkeyScope } from '@/object-record/record-table/types/TableOptionsHotkeyScope';
+import { useOpenObjectRecordsSpreadsheetImportDialog } from '@/object-record/spreadsheet-import/hooks/useOpenObjectRecordsSpreadsheetImportDialog';
+import { DropdownMenuHeader } from '@/ui/layout/dropdown/components/DropdownMenuHeader';
+import { DropdownMenuItemsContainer } from '@/ui/layout/dropdown/components/DropdownMenuItemsContainer';
+import { DropdownMenuSeparator } from '@/ui/layout/dropdown/components/DropdownMenuSeparator';
+import { useDropdown } from '@/ui/layout/dropdown/hooks/useDropdown';
+import { useDropdownContent } from '@/ui/layout/dropdown/hooks/useDropdownContent';
+import { useScopedHotkeys } from '@/ui/utilities/hotkey/hooks/useScopedHotkeys';
+import { ViewType } from '@/views/types/ViewType';
+
+export type RecordIndexOptionsContentId =
+  | 'viewSettings'
+  | 'fields'
+  | 'hiddenFields'
+  | 'recordGroups'
+  | 'hiddenRecordGroups'
+  | 'recordGroupFields'
+  | 'recordGroupSort';
+
+type ObjectOptionsDropdownContentProps = {
+  recordIndexId: string;
+  objectMetadataItem: ObjectMetadataItem;
+  viewType: ViewType;
+};
+
+export const ObjectOptionsDropdownContent = ({
+  viewType,
+  recordIndexId,
+  objectMetadataItem,
+}: ObjectOptionsDropdownContentProps) => {
+  const { currentContentId, onContentChange } =
+    useDropdownContent<RecordIndexOptionsContentId>();
+
+  const { closeDropdown } = useDropdown(OBJECT_OPTIONS_DROPDOWN_ID);
+
+  const { objectNamePlural } = useObjectNamePluralFromSingular({
+    objectNameSingular: objectMetadataItem.nameSingular,
+  });
+
+  useScopedHotkeys(
+    [Key.Escape],
+    () => {
+      closeDropdown();
+    },
+    TableOptionsHotkeyScope.Dropdown,
+  );
+
+  const { handleToggleTrashColumnFilter, toggleSoftDeleteFilterState } =
+    useHandleToggleTrashColumnFilter({
+      objectNameSingular: objectMetadataItem.nameSingular,
+      viewBarId: recordIndexId,
+    });
+
+  const { visibleBoardFields } = useObjectOptionsForBoard({
+    objectNameSingular: objectMetadataItem.nameSingular,
+    recordBoardId: recordIndexId,
+    viewBarId: recordIndexId,
+  });
+
+  const { viewGroupFieldMetadataItem } = useRecordGroups({
+    objectNameSingular: objectMetadataItem.nameSingular,
+  });
+
+  const { openObjectRecordsSpreasheetImportDialog } =
+    useOpenObjectRecordsSpreadsheetImportDialog(
+      objectMetadataItem.nameSingular,
+    );
+
+  const { progress, download } = useExportRecords({
+    delayMs: 100,
+    filename: `${objectMetadataItem.nameSingular}.csv`,
+    objectMetadataItem,
+    recordIndexId,
+    viewType,
+  });
+
+  return (
+    <>
+      {!currentContentId && (
+        <>
+          <DropdownMenuHeader StartIcon={IconList}>
+            {objectMetadataItem.labelPlural}
+          </DropdownMenuHeader>
+          {/** TODO: Should be removed when view settings contains more options */}
```

Comment:
we can remove this TODO as we don't have any other viewSettings for now?

Reply:
On Figma I've this:

<img width="262" alt="Screenshot 2024-11-19 at 9 00 30 AM" src="https://github.com/user-attachments/assets/66640f6d-108c-4f14-91f6-e66bbcf63801">